### PR TITLE
Add http script

### PR DESCRIPTION
**Pull request by noaco**

### CODE REVIEW

File: Scripts/script-http.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  - "false"
+  defaultValue: "false"
+- name: username
+  description: Base authentication
+- name: password
+  secret: true
+  description: Base authentication
+outputs:
+- contextPath: HttpRequest.Response
+  description: if The response for the http request
```
Comment (dorsha) [inline Scripts/script-http.yml]: No need for if, should be: `The response for the http request`

File: Scripts/script-http.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  - "false"
+  defaultValue: "true"
+- name: proxy
+  auto: PREDEFINED
+  predefined:
+  - "true"
+  - "false"
+  defaultValue: "false"
+- name: username
+  description: Base authentication
```
Comment (dorsha) [inline Scripts/script-http.yml]: Need to change the description like in the url, `The user for the http request`

File: Scripts/script-http.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+- name: password
+  secret: true
+  description: Base authentication
```
Comment (dorsha) [inline Scripts/script-http.yml]: Need to change the description like in the url, `The password for the http request`

File: Scripts/script-http.yml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  - DELETE
+  - PUT
+  description: The method for the http request
+- name: body
+  description: The body for the http request
+- name: url
+  required: true
+  description: The url for the http request
+- name: headers
+  description: The headers for the http request, in the format of 'key:value, key:value,
```
Comment (dorsha) [inline Scripts/script-http.yml]: the format example is with space, but in the code the split is `headersArr[i].split(':');`, so either fix the description or better trim the results

File: Scripts/script-http.yml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      ContentsFormat: formats.json,
+      EntryContext: {
+          'HttpRequest.Response': res
+      }
+  };
+
+
+
+type: javascript
+tags: []
```
Comment (dorsha) [inline Scripts/script-http.yml]: Should be have the `Utility` tag

File: Scripts/script-http.yml [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+  if (args.body) {
+      var body = (typeof args.body) === 'string' ? args.body : JSON.stringify(args.body);
+      requestObj.Body = body;
+  }
+
+  if (args.headers) {
+      var headersJs = {};
+      var headersArr = args.headers.split(',');
+      for ( i = 0 ; i < headersArr.length ; i++) {
```
Comment (dorsha) [inline Scripts/script-http.yml]: remove the space after the `(`

File: Scripts/script-http.yml [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  id: http
+  version: -1
+name: http
+script: |+
+  var requestObj = {
+      Method: args.method,
+  };
+
+  if (args.username) {
+      requestObj.Username =args.username;
```
Comment (dorsha) [inline Scripts/script-http.yml]: add space after the `=`

File: Scripts/script-http.yml [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+  }
+
+  if (args.username) {
+      requestObj.Password =args.password;
```
Comment (dorsha) [inline Scripts/script-http.yml]: add space after the `=`

File: Scripts/script-http.yml [hunk 9]
```diff
@@ -0,0 +1,90 @@
+commonfields:
+  id: http
+  version: -1
+name: http
```
Comment (dorsha) [inline Scripts/script-http.yml]: script description is missing, should have good explanation about the script and some examples.

**Comment by noaco** [comment]
@dorsha please review again. thanks.

**Pull request merged**